### DISCUSSION TITLE
How to get the token from response header or local storage?

### DISCUSSION THREAD

**Discussion by wanghaomiao27** [cypress-io/cypress#28328] [General]
I have a scenario:when I make a login request,I get a token,the token will be generated at response headers,or you can find it at local storage.I want to get the token,because for next request ,it shall use this token.Could anyone give me some detailed code snippets to 
 get this token and for next request use?Belows are some screenshots for this;
![image](https://github.com/cypress-io/cypress/assets/20040189/8926fc79-40e2-497a-b7f7-7ec5f7a1be08)
![image](https://github.com/cypress-io/cypress/assets/20040189/d54fdd70-4c98-42d0-aa28-2bc1d9b4aab3)
it('登录请求',()=>{
        //cy.wrap(localStorage.getItem())
            cy.request({
                method:'POST',
                url:'http://wolf.test.datatist.cn/analyzer/analyzer/account/login.do',
                headers:{"Project-Id":"Jevf4ghaKT091r5E",
                        "Content-Type":"application/json"
                },
                failOnStatusCode:false,
                body:{
                    "password": "abcd1234",
                    "rememberMe": true,
                    "scope": "",
                    "username": "analyzer@datatist.com"
                }
            }).then((response)=>{
                expect(response.status).to.eq(200)
            }
    )

**Comment by adnanerlansyah403** [comment]
you can use the query cy.window() to get the current localStorage that already exist on your browser and wrap all your code on that callback window(). the details of code : 
```
cy.window().its("localStorage").invoke('getItem', 'authorization').then(token => {
  // use the data token from callback to put it on the headers for the request signin
})
```
